hudi-common/src/main/java/org/apache/hudi/avro/HoodieAvroUtils.java

```diff
@@ -214,6 +218,20 @@ private static GenericRecord rewrite(GenericRecord record, Schema schemaWithFiel
     return newRecord;
   }
 
+  /*
+  This function takes the union of all the fields except hoodie metadata fields
+   */
+  private static List<Field> getAllFieldsToWrite(Schema oldSchema, Schema newSchema) {
+    Set<Field> allFields = new HashSet<>(oldSchema.getFields());
+    List<Field> fields = new ArrayList<>(oldSchema.getFields());
```

Comment:
would a `LinkedHashSet` eliminate the need for a set and a list?

Reply:
Yes it helps. Thank you for suggesting this :)